dkist/wcs/tests/test_models.py

```diff
@@ -46,6 +46,19 @@ def test_generate_celestial_unitless():
     assert u.allclose(shift1.offset, 0)
 
 
+def test_varying_transform_no_lon_pole_unit():
```

Comment:
If you prefer, I can remove this test and add the unit check to the test below and remove the lon_pole argument in the `VaryingCelestialTransform` call.

I can also add a test for `generate_celestial_transform` instead? Or both?